File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/UserInHomeCommand.java
package wpam.mobile_client.protocol;

import java.util.ArrayList;

public final class UserInHomeCommand extends Command
{
    public UserInHomeCommand()
    {
        super(10, CommandType.USER_OUT_OF_HOME);
    }
    public final void close()
    {
        super.close();
    }

    @Override
    public ArrayList<Integer> getFrameBytes()
    {
        initializeDataSize();

        ArrayList<Integer> frame = getHeader();
        frame.add(commandType_.getValue());

        return frame;
    }
    @Override
    public String getName()
    {
        return new String("UserInHomeCommand");
    }

    @Override
    public void accept(CommandVisitor visitor)
    {
        visitor.visit(this);
    }

    @Override
    protected void initializeDataSize()
    {
        setDataSize(1);
    }
}

File: server/src/protocol/UserInHomeCommand.cpp
#include "UserInHomeCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;

UserInHomeCommand::UserInHomeCommand()
        : Command(10, CommandType::USER_IN_HOME)
{}

UserInHomeCommand::~UserInHomeCommand()
{}

vector<uint8_t> UserInHomeCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

string UserInHomeCommand::getName()
{
    return string("UserinHomeCommand");
}

void UserInHomeCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void UserInHomeCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}
